### PR TITLE
Steps towards #246

### PR DESCRIPTION
**Pull request by tBuLi**
This makes some steps to completing #246 by updating the API docs section to the latest changes. Although not nearly complete, it is a small improvement I can make with the time I have right now.

### CODE REVIEW

File: docs/api_structure.rst [hunk 1]
```diff
@@ -10,75 +10,83 @@ Fitting 101
...
+----------
 
-Now instead, we want to call the minimizer directly. We first define a custom
-objective function (actually just a chi squared)::
+Objectives wrap both the Model and the data supplied, and expose only the free
+parameters of the model to the outside world.
+When called they must return a scalar. This scalar will be *minimized*, so when
+you need something maximized, be sure to add a negation in the right place(s).
+They can be called by using the parameter names as keyword arguments, or with a
+list of parameter values. The latter is there because this is how ``scipy``
```
Comment (pckroon) [inline docs/api_structure.rst]: list of parameter values: what order?

File: docs/api_structure.rst [hunk 2]
```diff
@@ -10,75 +10,83 @@ Fitting 101
...
+likes it.
+Be sure to inherit from the abstract base class(es) so you're sure you define
+all the methods that are expected of an objective. Similar to the models, they
+come in three types: ``BaseObjective``, ``GradientObjective`` and
+``HessianObjective``. These must implement ``__call__``, ``eval_jacobian`` and
+``eval_hessian`` respectively.
+
+When defining a new objective, it is best to inherit from ``HessianObjective``
+and to define all three if possible. When feeding a model that is only callable
```
Comment (pckroon) [inline docs/api_structure.rst]: explicitly: a BaseModel or GradientModel

File: docs/api_structure.rst [hunk 3]
```diff
@@ -10,75 +10,83 @@ Fitting 101
...
-    def chi_squared(a, b):
-        return np.sum((ydata - f(xdata, a, b))**2)
+Last in the chain are the minimizers. They are provided with a function to
+minimize (the objective) and the :class:`~symfit.core.argument.Parameter` s as
+a function of which the objective should be minimized. Note that once again
+there are different base classes for minimizers that take e.g. bounds or
+support gradients. Their :meth:`~symfit.core.minimizers.BaseMinimizer.execute`
+method takes the metaparameters for the minimization.
+Again, be sure to inherit from the appropriate base class(es) if you're
+implementing your own minimizer to make sure all the expected methods are there.
```
Comment (pckroon) [inline docs/api_structure.rst]: ... to make sure all the expected methods are there, and Fit can figure out what it can do.

**Comment by DuckNrOne** [comment]
It's in every case a step in the right direction 👍 
@pckroon I'll look from time to time in the docs. I think, that a link to the specific file would be a great help.

**Comment by tBuLi** [comment]
@DuckNrOne Can you check if this answers your questions, and if not what more you'd need to know?

**Comment by DuckNrOne** [comment]
With this API structure it's possible for me to understand your structure. I would merge this 👍 

As you said, if I have an additional question, I'll open an issue. Thanks guys!

**Pull request merged**